integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+        body: null
+      },
+      userName,
+      statusCode
+    ).then((response) => {
+       if(statusCode==null || statusCode==200) {
+	  // API returns devices: [ groupname: ..., devicename: ..., saltId, ..., Group.groupName: ... ]
+	  // sort users and expected users to ensure order is the same
+          var users=sortArrayOn(response.body.rows, 'username');
+          expectedUsers=sortArrayOn(expectedUsers, 'username');
+	  expect(users.length).to.equal(expectedUsers.length);
+
+          for (var index=0; index < expectedUsers.length; index++) {
+		  expect(users[index].id).to.equal(expectedUsers[index].id);
+		  expect(users[index].username).to.equal(expectedUsers[index].username);
+		  expect(users[index].relation).to.equal(expectedUsers[index].relation);
+		  expect(users[index].admin).to.equal(expectedUsers[index].admin);
+		  expect(users[index].email).to.equal(expectedUsers[index].email);
+	  }
+       };
+  });
+});
+
+Cypress.Commands.add(
+  "apiAddUserToDevice",
+  (deviceAdminUser: string, userName: string, device: string, admin: boolean = false, statusCode: number = 200) => {
+    logTestDescription(
+      `${deviceAdminUser} Adding user '${userName}' to device '${device}'`,
+      { user: userName, device }
+    );
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "POST",
+        url: v1ApiPath("devices/users"),
+        body: {
+          deviceId: getCreds(device).id,
+          admin: admin,
+          username: getTestName(userName)
+        }
+      },
+      deviceAdminUser,
+      statusCode
+    );
+  }
+);
+
+Cypress.Commands.add(
+  "apiRemoveUserFromDevice",
+  (deviceAdminUser: string, userName: string, device: string, statusCode: number = 200) => {
+    logTestDescription(
+      `${deviceAdminUser} Removing user '${userName}' to device '${device}'`,
+      { user: userName, device }
+    );
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "DELETE",
+        url: v1ApiPath("devices/users"),
+        body: {
+          deviceId: getCreds(device).id,
+          admin: "false",
+          username: getTestName(userName)
+        }
+      },
+      deviceAdminUser,
+      statusCode
+    );
+  }
+);
+
+
+
+
+Cypress.Commands.add("apiUploadRecording", (cameraName, id) => {
+  const recordingsUrl = v1ApiPath('recordings');
+  const deviceName = getTestName(cameraName);
+
+  const data = {
+       type: 'thermalRaw'
+  };
+
+  const fileName="bird_"+id+".cptv";
+  const uniqueName=deviceName+"_"+id;
+  const aliasName=deviceName+"_a"+id;
+
+
+  const creds = getCreds(cameraName);
+  uploadFileRequest(fileName, uniqueName, aliasName, recordingsUrl, data, creds)
+
+});
+
+
+Cypress.Commands.add("apiUploadEvent", (cameraName) => {
+  const eventsUrl = v1ApiPath('events');
+  const deviceName = getTestName(cameraName);
+  const creds = getCreds(cameraName);
+
+  const data = {
+       description: {
+	   type: 'throttle'
+       },
+       dateTimes: ['2017-11-13T00:47:51.160Z']
```

Comment:
should this be a nullable parameter and if null set to datetime.now?

Reply:
Yes.  Noted. This existing function will be rewritten to allow flexible testing when Events tests are written.

Reply:
This was a duplicate function from browser tests.  Removed and browser tests pointed at the correct api function..

Reply:
Function deleted.  Duplicate function from browser test code.  Browser tests now pointed at the correct version of this function and this function deleted.